docs/proposals/scheduledjob.md

```diff
@@ -0,0 +1,163 @@
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# ScheduledJob Controller
+
+## Abstract
+
+A proposal for implementing a new controller - ScheduledJob controller - which
+will be responsible for managing time based jobs, namely:
+* once at a specified point in time,
+* repeatedly at a specified point in time.
+
+There is already a discussion regarding that particular subject:
+* Distributed CRON jobs [#2156](https://github.com/kubernetes/kubernetes/issues/2156)
+
+There are also similar solutions available, already:
+* [Mesos Chronos](https://github.com/mesos/chronos)
+* [Quartz](http://quartz-scheduler.org/)
+
+
+## Use Cases
+
+1. Be able to schedule a job execution at a given point in time.
+1. Be able to create a periodic job, eg. database backup, sending emails.
+
+
+## Motivation
+
+ScheduledJobs are needed for performing all time related actions, namely backups,
+report generation and alike.  Each of these tasks should be allowed to perform
+repeatedly (once a day/month, etc.) or once at a given point in time.
+
+
+## Implementation
+
+ScheduledJob controller relies heavily on the [Job Controller API](job.md)
+for running actual jobs, on top of which it adds information regarding the date
+and time part according to ISO8601 format.
+
+The new `ScheduledJob` object will have the following content:
+
+```go
+// ScheduledJob represents the configuration of a single scheduled job.
+type ScheduledJob struct {
+    TypeMeta
+    ObjectMeta
+
+    // Spec is a structure defining the expected behavior of a job, including the schedule.
+    Spec ScheduledJobSpec
+
+    // Status is a structure describing current status of a job.
+    Status ScheduledJobStatus
+}
+
+// ScheduledJobList is a collection of scheduled jobs.
+type ScheduledJobList struct {
+    TypeMeta
+    ListMeta
+
+    Items []ScheduledJob
+}
+```
+
+`ScheduledJobSpec` structure is defined to contain all the information how the actual
+job execution will look like, including the `JobSpec` from [Job Controller API](job.md)
+and the schedule in ISO8601 format.  This implies that each ScheduledJob execution
+will be created from the JobSpec actual at a point in time when the execution will
+be started.
+
+```go
+// ScheduledJobSpec describes how the job execution will look like and when it will actually run.
+type ScheduledJobSpec struct {
+
+    // Template is the object that describes the job that will be created when
+    // executing a ScheduledJob.
+    Template *JobSpec
+
+    // Selector is a label query over created jobs, sorted and distinguished by CreationTimestamp.
+    Selector map[string]string
+
+    // Schedule contains the schedule in ISO8601 format, eg.
+    // - 2015-07-21T14:00:00Z - represents date and time in UTC
+    // - R/2015-07-21T14:00:00Z/P1D - represents endlessly repeating interval (1 day), starting from given date
+    Schedule string
+
```

Comment:
Right, those where my feelings as well, though I know nothing about mergeable strings and need to read about it a bit. But again, when updating you'll just modify the current schedule so that it matches your new expectations and submit the new one which will replace the old one.